src/tools/kata-ctl/src/arch/x86_64/mod.rs

```diff
@@ -7,11 +7,47 @@
 pub use arch_specific::*;
 
 mod arch_specific {
-    use anyhow::Result;
+    use anyhow::{anyhow, Result};
+    use crate::check;
+
+    const PROC_CPUINFO: &str = "/proc/cpuinfo";
+    const CPUINFO_DELIMITER: &str = "\nprocessor";
+    const CPUINFO_FLAGS_TAG: &str = "flags";
+    const CPU_FLAGS_INTEL: &'static [&'static str] = &["lm", "sse4_1", "vmx"];
+    const CPU_ATTRIBS_INTEL: &'static [&'static str] = &["GenuineIntel"];
+
+    // check cpu
+    fn check_cpu() -> Result<()> {
+        println!("INFO: check CPU: x86_64");
+
+        let cpu_info = check::get_single_cpu_info(PROC_CPUINFO, CPUINFO_DELIMITER)?;
+
+        let cpu_flags = check::get_cpu_flags(&cpu_info, CPUINFO_FLAGS_TAG)
+            .map_err(|e| anyhow!("Error parsing CPU flags, file {:?}, {:?}", PROC_CPUINFO, e))?;
+
+        // perform checks
+        // TODO: Perform checks based on hypervisor type
+        // TODO: Add more information to output (see kata-check in go tool); adjust formatting
+        let missing_cpu_attributes = check::check_cpu_attribs(&cpu_info, CPU_ATTRIBS_INTEL)?;
+        if missing_cpu_attributes.len() > 0 {
+            eprintln!("WARNING: Missing CPU attributes {:?}", missing_cpu_attributes);
+        }
+        let missing_cpu_flags = check::check_cpu_flags(&cpu_flags, CPU_FLAGS_INTEL)?;
+        if missing_cpu_flags.len() > 0 {
+            eprintln!("WARNING: Missing CPU flags {:?}", missing_cpu_flags);
+        }
```

Comment:
The `log` crate is a good option for handling logging with different verbose levels, say 'warn', 'error', 'debug', etc. It can also be exposed by a CLI parameter for users to decide the logging level.

Reply:
@likebreath - thanks. I've raised an issue to use the Kata logging crate (which uses `slog`):

- https://github.com/kata-containers/kata-containers/issues/5401